evalml/utils/woodwork_utils.py

```diff
@@ -8,122 +8,112 @@
+
+def _convert_woodwork_types_wrapper():
+    """Placeholder so we don't have to edit every file in the repo yet."""
+
+
+def _retain_custom_types_and_initalize_woodwork(old_logical_types, new_dataframe, ltypes_to_ignore=None):
     """
     Helper method which will take an old Woodwork data structure and a new pandas data structure and return a
     new data structure that will try to retain as many logical types from the old data structure that exist in the new
     pandas data structure as possible.
 
     Arguments:
-        old_woodwork_data (ww.DataTable): Woodwork data structure to use
-        new_pandas_data (pd.DataFrame): Pandas data structure
-        ltypes_to_ignore (list): List of Woodwork logical types to ignore. Columns from the old DataTable that have a logical type
-        specified in this list will not have their logical types carried over to the new DataTable returned
+        old_logical_types (Dict): Logical types to try to retain.
+        new_dataframe (pd.DataFrame): Pandas data structure
+        ltypes_to_ignore (list): List of Woodwork logical types to ignore. Columns from the old DataFrame that have a logical type
+        specified in this list will not have their logical types carried over to the new DataFrame returned
 
     Returns:
-        A new DataTable where any of the columns that exist in the old input DataTable and the new DataFrame try to retain
+        A new DataFrame where any of the columns that exist in the old input DataFrame and the new DataFrame try to retain
         the original logical type, if possible and not specified to be ignored.
     """
-    if isinstance(old_woodwork_data, ww.DataColumn):
-        if str(new_pandas_data.dtype) != old_woodwork_data.logical_type.pandas_dtype:
-            try:
-                return ww.DataColumn(new_pandas_data, logical_type=old_woodwork_data.logical_type)
-            except (ValueError, TypeError):
-                pass
-        return ww.DataColumn(new_pandas_data)
-
-    retained_logical_types = {}
+    if isinstance(new_dataframe, pd.Series):
+        return ww.init_series(new_dataframe, old_logical_types)
     if ltypes_to_ignore is None:
         ltypes_to_ignore = []
-    col_intersection = set(old_woodwork_data.columns).intersection(set(new_pandas_data.columns))
-    logical_types = old_woodwork_data.logical_types
-    for col in col_intersection:
-        if logical_types[col] in ltypes_to_ignore:
-            continue
-        if str(new_pandas_data[col].dtype) != logical_types[col].pandas_dtype:
-            try:
-                new_pandas_data[col].astype(logical_types[col].pandas_dtype)
-                retained_logical_types[col] = old_woodwork_data[col].logical_type
-            except (ValueError, TypeError):
-                pass
-    return ww.DataTable(new_pandas_data, logical_types=retained_logical_types)
+    col_intersection = set(old_logical_types.keys()).intersection(set(new_dataframe.columns))
```

Comment:
Nothing makes me happier than when set intersection or symmetric_difference is used.